### DISCUSSION TITLE
Apollo Server & Nexus - context invalid when querying relations

### DISCUSSION THREAD

**Discussion by erdfern** [prisma/prisma#9570] [Q&A]
Hey, I've got a problem. I'm using prisma client, nexus and the nexus-prisma plugin, as well as apollo-server.

I've got a simple Nexus resolver:
```ts
    t.list.field('projects', {
      type: 'Project',
      async resolve(_root, _args, ctx) {
        return await ctx.db.project.findMany({include: {albums: true}});
      },
    })
```

Which, when executed with the following query (in Apollo Studio):
```gql
query GetProjects {
  projects {
    name
    albums {
      id
    }
  }
}
```
... returns this error: `The GraphQL context.prisma value is not an instance of the Prisma Client (class reference for check imported from @prisma/client).` The database (postgres) is populated with both `project`s and `album`s, and they are shown correctly when introspecting the db in the Data Browser tab of Prisma Studio.

What's unclear to me is
1. Why the query works just fine if I don't query for the relation (if I omit `albums { id})
2. Why the query works just fine in Prisma Studio, but not in Apollo Studio:
    ```ts
    // in the Prisma Studio query console
    await prisma.project.findMany({
      include: {albums: true}
    })

   // result
   [
      {
        "id": "ckuavlkkq0000a5ab8t4pm2ux",
        "name": "p0",
        "albums": [
          {
            "id": "ckuavo5ia0000moab929uhwn9",
            "projectId": "ckuavlkkq0000a5ab8t4pm2ux"
          }]}]
    ```
The models are roughly defined like this:
```gql
model Project {
  id   String @id @default(cuid())
  name String @unique

  albums Album[]
}

model Album {
  id        String   @id @default(cuid())
  Project   Project? @relation(fields: [projectId], references: [id])
  projectId String?

  images Image[]
}

...
```
My context is defined as such:
```ts
import {PrismaClient} from '@prisma/client';
import {db} from './db';

export interface Context {
  db: PrismaClient;
}

export const context = {
  db,
};
```
And then registered in the makeSchema function and the Apollo Server:
```ts
//...
contextType: {module: join(__dirname, './context.ts'), export: 'Context'},
//...

import {ApolloServer} from 'apollo-server';
import {context, schema} from './api';

export const server = new ApolloServer({schema, context});
```

Any pointers in the right direction are greatly appreciated! :pensive:

**Comment by erdfern** [comment]
Uhh... So, I renamed the `db` property of my `context`object to `prisma`. That fixed it.

Apparently, apollo was keen on that. (it'd be nice if that was mentioned anywhere).

¯\_(ツ)_/¯

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
